nalgebra-sparse/src/cs.rs

```diff
@@ -679,14 +666,22 @@ where
                 );
 
                 // sort values if they exist
-                if !value_refs.is_empty() {
+                if let Some(values) = values_option.as_mut() {
+                    if values.len() == 0 {
+                        panic!(
+                            "Internal error: Sorting currently not supported if values are empty."
+                        );
+                    }
```

Comment:
I don't understand this panic. Certainly we need to be able to handle this case, as it can easily happen based on user input. Also, what do we want to prevent here? It seems to me that the code should do the correct thing even in the case where `values` is empty?

Reply:
This could be an error indeed, but I am not sure if we should let the method do the correct thing regardless of empty `values` as I thought that we decided to have this restriction for now (see [comment](https://github.com/dimforge/nalgebra/pull/1015#discussion_r766490039)) .

 Otherwise we should also remove panic for the case [when `values` is None](https://github.com/dimforge/nalgebra/pull/1015/commits/bd864b84334a5496bdfc72a219570c5fb9c76831#diff-ad9fa9a189df6658e5fee89093cd2efba6292df75401887d4c3e336a6fca2e4bR563) .
 But for this case you said that it would be OK to panic, so I am also a bit confused when it is OK to panic and when not.
 
 I suppose we can chose between turning both cases into errors and test them or letting current method do the correct thing regardless of whether `values` is provided/non-empty or not?